### PR TITLE
WINC-1340: Build kube-proxy from kube v1.31 + Master submodule update 10 07

### PR DESCRIPTION
**Pull request by saifshaikh48**
Also includes 
- Bump kube-proxy build to use k8s v1.31 tag

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,8 +7,8 @@ WMCO_VERSION ?= 10.18.0
 
 # *_GIT_VERSION are the k8s versions. Any update to the build line could potentially require an update to the sed
 # command in generate_k8s_version_commit() in hack/update_submodules.sh
-KUBELET_GIT_VERSION=v1.31.1+1892e4d
-KUBE-PROXY_GIT_VERSION=v1.30.0
+KUBELET_GIT_VERSION=v1.31.1+8540bd4
+KUBE-PROXY_GIT_VERSION=v1.31.1
 CONTAINERD_GIT_VERSION=v1.7.20
```
Comment (jrvaldes) [inline Makefile:12]: @saifshaikh48   should this bump containerd as well

https://github.com/containerd/containerd/releases/tag/v1.7.22

Comment (saifshaikh48) [reply Makefile:12]: hack script did not generate commit for containerd. from the submodule dir the version is v1.7.20 after running

**Comment by saifshaikh48** [comment]
/test vsphere-disconnected-e2e-operator 

timeout after waiting to get infra lease for 1.5hrs

**Comment by jrvaldes** [comment]
azure-e2e-operator depends on:

- https://github.com/openshift/windows-machine-config-operator/pull/2471

**Comment by jrvaldes** [comment]
@openshift/openshift-team-windows-containers  consider overriding the azure-e2e-operator job. 

  `azure-e2e-upgrade`  [passed](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_windows-machine-config-operator/2449/pull-ci-openshift-windows-machine-config-operator-master-azure-e2e-upgrade/1844086625764642816) which validates the kube 1.31 bump in Azure.

**Pull request merged**